blocksuite/affine/block-attachment/src/embed.ts

```diff
@@ -71,21 +71,22 @@ export class AttachmentEmbedService extends Extension {
     return this.configs.values();
   }
 
-  constructor(
-    private readonly std: BlockStdScope,
-    private readonly configs: Map<string, AttachmentEmbedConfig>
-  ) {
+  get configs(): Map<string, AttachmentEmbedConfig> {
+    return this.std.get(AttachmentEmbedConfigMapIdentifier);
+  }
+
+  constructor(private readonly std: BlockStdScope) {
     super();
   }
 
   static override setup(di: Container) {
     di.addImpl(AttachmentEmbedConfigMapIdentifier, provider =>
       provider.getAll(AttachmentEmbedConfigIdentifier)
     );
-    di.addImpl(AttachmentEmbedProvider, AttachmentEmbedService, [
-      StdIdentifier,
-      AttachmentEmbedConfigMapIdentifier,
-    ]);
+    di.addImpl(
+      AttachmentEmbedProvider,
+      new AttachmentEmbedService(di.provider().get(StdIdentifier))
```

Comment:
This is not right, why don't use
```ts
di.addImpl(
      AttachmentEmbedProvider,
      AttachmentEmbedService,
      [StdIdentifier]
)
```

Your implementation will throw error if this extension is registered before stdIdentifier.